app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgSendQueue.java

```diff
@@ -47,23 +47,13 @@
     @Column(name = "operation_type")
     public String operation_type;
 
-/*
-    public static List<BgSendQueue> queue() {
-        return new Select()
-                .from(BgSendQueue.class)
-                .where("success = ?", false)
-                .orderBy("_ID asc")
-                .limit(20)
-                .execute();
-    }
-*/
     public static List<BgSendQueue> mongoQueue() {
         return new Select()
                 .from(BgSendQueue.class)
                 .where("mongo_success = ?", false)
                 .where("operation_type = ?", "create")
-                .orderBy("_ID desc")
-                .limit(30)
+                .orderBy("_ID asc")
```

Comment:
This is the first time I see an advantage of setting up NightScout for adults using mongo just as data backup ;)
With mongo-uploads I guess it is more a weak gsm connection that screws up things than mongolab. But a few days a go when we talked about upping the amount of data, I also upped the number and lowered the repitition time. After some uploads it always ended in failed uploads for a short period of time.
It might have been only the upload of one value that actually failed, but it always causes the whole batch to keep being marked not uploaded. :/
As long as these cases can happen, I'm strongly in favour of a solution that sets priority to the most recent "live" value to be uploaded. (Even though I personally just upload once in a while and on wifi and would personally profit from a faster burst upload.)